bims/api_views/taxon.py

```diff
@@ -74,15 +79,37 @@ def get(self, request, pk, format=None):
         for rank in taxonomic_rank:
             data.update(rank)
         common_names = []
+        results = []
 
         # Common name
         if taxon.vernacular_names.exists():
             common_names = list(
-                taxon.vernacular_names.all().values_list('name', flat=True))
+                taxon.vernacular_names.filter(language='eng').values_list('name', flat=True))
         if len(common_names) == 0:
-            data['common_name'] = 'Unknown'
+            vernacular_names = get_vernacular_names(taxon.gbif_key)
```

Comment:
Maybe also check if the taxon has gbif key.

Reply:
The feature `get_vernacular_names` returns None if gbif key doesn't exist.